agent/structs/config_entry_gateways.go

```diff
@@ -975,6 +1048,22 @@ func (e *BoundAPIGatewayConfigEntry) IsSame(other *BoundAPIGatewayConfigEntry) b
 		}
 	}
 
+	for key, refs := range e.Services {
+		if _, ok := other.Services[key]; !ok {
+			return false
+		}
+
+		if len(refs) != len(other.Services[key]) {
+			return false
+		}
+
+		for idx, ref := range refs {
+			if !ref.IsSame(&other.Services[key][idx]) {
```

Comment:
Thinking that this could panic if `other.Services[key]` doesn't exist. Can you add a guard for it?

Reply:
we already check for `other.Services[key]` in the first guard clause

Reply:
oh that's a good point, let me add that because we should include that check just to be sure

Reply:
added on line 1031